Include documentation for including an R method (#99)
NERC-CEH/data-science-toolbox: about_this_project/using_r.md

# Including R Methods

For methods developed in `R`, documentation will often take the form of an `R Markdown` document, `Quarto` document, or package `vignette`. For inclusion in the Environmental Data Science Toolbox, however, methods are presented as Jupyter `.ipynb` notebooks, which can be executed using an R kernel. The approach is used so methods developed in `R`, `Python` and other languages are all visible in the same interface, promoting cross-language method discovery and knowledge sharing.

> For methods developed in R the normal approach to documenting the method is to have an rmarkdown or vignette. For inclusion into the toolbox we need an Jupyter .ipynb file, which can be run with an R kernel. The approach is used to allow having methods developed inR and Ypthon and other languages all visible in a similar interface, promoting cross-language method discovery and knowledge sharing.

Reviewer: ```suggestion
For methods developed in R the normal approach to documenting the method is to have an rmarkdown or vignette. For inclusion into the toolbox we need a Jupyter .ipynb file, which can be run with an R kernel. The approach is used to allow having methods developed in R and Python and other languages all visible in a similar interface, promoting cross-language method discovery and knowledge sharing.
```

> For methods developed in R the normal approach to documenting the method is to have an rmarkdown or vignette. For inclusion into the toolbox we need a Jupyter .ipynb file, which can be run with an R kernel. The approach is used to allow having methods developed in R and Python and other languages all visible in a similar interface, promoting cross-language method discovery and knowledge sharing.

Author: ```suggestion
```

Accidental repeat of above.

For reproducibility, best practice is to include a `DESCRIPTION` file or to use `renv`. This will easily allow users to recreate the necessary environment for running the notebook.

## 2. Install and register an R Jupyter kernel.

Register the R environment you're in as an R kernel for use by the notebook. To do this we use `IRkernel`, which you'll have to install through `install.packages('IRkernel')`.

Register the R kernel:

```r
IRkernel::installspec(
    name = "notebookname-r",
    displayname = "R - notebookname"
)
```

This is normally a one-off setup step on each machine.

## 3. Create an .ipynb file and select the R kernel.

While several approaches exist for converting R markdown/quarto documents into Jupyter .ipynb notebooks, we recommend simply starting from fresh with the following [notebook template](https://github.com/NERC-CEH/data-science-toolbox/blob/main/notebook_guidance/template_notebook.ipynb) and copying across code/text. Ultimately, this is more efficient than troubleshooting issues with a direct conversion and it allows you to test the cells all run as expected. 

> Open the [notebook template](https://github.com/NERC-CEH/data-science-toolbox/blob/main/notebook_guidance/template_notebook.ipynb) file in a interface such as VS Code with the Jupyter extension, JupyterLab, or another Jupyter-compatible interface. Select the R kernel registered in the previous step:

Reviewer: ```suggestion
Open the [notebook template](https://github.com/NERC-CEH/data-science-toolbox/blob/main/notebook_guidance/template_notebook.ipynb) file in an interface such as VS Code with the Jupyter extension, JupyterLab, or another Jupyter-compatible interface. Select the R kernel registered in the previous step:
```

Reviewer 2: This looks good - glad to see support for R being included. Have accepted suggestions on typo correcting. Will send for re-review.
